lib/event_store_client/mapper/encrypted.rb

```diff
@@ -48,7 +49,34 @@ def serialize(event)
       #   which key should be used as an identifier.
       # *Returns*: Specific event class with all data decrypted
 
-      def deserialize(_event); end
+      def deserialize(event)
+        metadata = serializer.deserialize(event.metadata)
+        encryption_schema = serializer.deserialize(event.metadata)['encryption']
+        data = serializer.deserialize(event.data)
+        decryptor = DataDecryptor.new(
+          data: data,
+          schema: encryption_schema,
+          repository: key_repository
+        )
+        decryptor.call
+
+        event_class =
+          begin
+            Object.const_get(event.type)
+          rescue NameError
+            EventStoreClient::DeserializedEvent
+          end
+
+        data = decryptor.decrypted_data
```

Comment:
I would move `decryptor.decrypted_data` to line 76 it's more performant than assigning it to the var(data) and doing nothing with this var.